packages/compiler/test/checker/references.test.ts

```diff
@@ -595,5 +595,39 @@ describe("compiler: references", () => {
         `,
         ref: "testOp::parameters.select",
       }));
+
+    it("emits a diagnostic when referencing a non-existent meta type property", async () => {
```

Comment:
Can we separate the 2 diagnostics in 2 tests:
- `ModelProperty doesn't have meta property type` as this one is a temporary error until the below resolution is solved
-  `Model doesn't have meta property foo` this one is I believe something that is completly correct to complain about

Reply:
Will do

Reply:
Filed issue https://github.com/microsoft/typespec/issues/2213 to address the underlying problem.

Reply:
I mean can we just have the 2 diagnostics in separate “it” and link the one that is a temporary block to the issue you filed

that way we can remove it

Reply:
One thing to clarify: the message is reporting that a particular property is being requested but it doesn't exist.  In this case, I'm using `B.a::type` which currently cannot be resolved due to the underlying bug.  In the future when the underlying bug is fixed, a diagnostic will no longer get raised for this meta property because it will exist.  Does that make sense?

Reply:
@timotheeguerin any other thoughts here?

Reply:
Yeah that is my point right now there is those 2 diagnostics but in the future there should only be one(the 2nd in the list)

```json
 {
          code: "invalid-ref",
          message: `Model doesn't have meta property foo`,
        },
```

so my point was just to separate out the 2 scenarios here in 2 test so ew can just delete the test that check for
```json
{
          code: "invalid-ref",
          message: `ModelProperty doesn't have meta property type`,
        },
``` 

with both those being in the some test it makes it seems like that error is an expected behavior of the compiler and just a temporary bug